backend/app/services/alert_engine.py

```
@@ -255,15 +264,46 @@ def evaluate_realtime(
+      occ_cond = rule.get("occupancy_pct")
+      if occ_cond and zone_id and zone_counts is not None:
+        try:
+          zone_doc = zones_col.find_one({"camera_id": camera_id, "zone_id": zone_id})
+          cap = zone_doc.get("capacity") if zone_doc else None
+          if cap is not None and int(cap) > 0:
+            occ_pct = 100.0 * cnt / int(cap)
+            for op, val in occ_cond.items():
+              try:
+                v = float(val)
+              except Exception:
+                continue
```

Ruff (0.15.1)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)